.github/workflows/ci-backend.yml

```
@@ -0,0 +1,26 @@
+name: Java CI with Gradle
+
+on:
+  pull_request:
+    branches:
+      - main
+      - develop
```

트리거 몇 가지만 더 추가하는게 어떨까요? PR만이 아니라 브랜치에 대한 직접적인 커밋 푸시에 대해서도 동작하고, 아래의 경우에 대해서도 동작하게요.

참고: [공식문서](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)

타입 설명

- opened: PR이 열렸을 때
- ready_for_review: draft로 열렸던 PR을 리뷰 가능 상태로 수정했을 때
- reopened: 닫혔던 PR을 다시 열었을 때
- synchronize: PR로 올라온 브랜치에 대한 커밋 추가

```suggestion
on:
  push:
    branches:
      - main
      - develop
  pull_request:
    branches:
      - main
      - develop
    types: [opened, ready_for_review, reopened, synchronize]
```

---

push에 대해선 동의합니다.

다만, 제가 테스트해보았을 때는 기본적으로 동일 브랜치의 pr에 새롭게 push할 때마다 workflow 트리거가 동작하긴 하더라구요. `types`가 없어도 될 것 같은 느낌? 

요건 어떤가요?
```yml
on:
  push:
    branches:
      - main
      - develop
  pull_request:
    branches:
      - main
      - develop
```

---

넵. 예전에 `types`가 필요해서 추가했었던건데, 없어도 동작한다면 일단 빼도 괜찮을 것 같아요.

---

우선은 위처럼 추가해봤습니다. 이후에 types 필요성이 생기면 추가 또는 issue 생성하면 될 것 같습니다 :)